# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGL2ContextTransformFeedback.cpp
# Commit: 71c122ac0ca7
# Full Hash: 71c122ac0ca73391866b1ef19f4f82bc2d28568b
# Author: David Parks <davidp99@gmail.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Initial out-of-process WebGL implementation. r=mccr8,handyman
#   
#   Splits WebGLContext into ClientWebGLContext and HostWebGLContext.  The Client enables the JS-control of a WebGL context in a content procecss while the Host executes the WebGL graphics operations (via a WebGLContext that maintains much of the existing code) in the compositor process.  At this point, the cross-process behavior is disabled -- this series of patches is an incremental step toward that final goal.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54018
# ==============================================================================

diff -r 5cbf5bb11c58 -r 71c122ac0ca7 dom/canvas/WebGL2ContextTransformFeedback.cpp
--- a/dom/canvas/WebGL2ContextTransformFeedback.cpp	Wed Jan 08 21:52:03 2020 +0000
+++ b/dom/canvas/WebGL2ContextTransformFeedback.cpp	Wed Jan 08 22:19:14 2020 +0000
@@ -110,7 +110,7 @@
 }
 
 void WebGL2Context::TransformFeedbackVaryings(
-    WebGLProgram& program, const dom::Sequence<nsString>& varyings,
+    WebGLProgram& program, const nsTArray<nsString>& varyings,
     GLenum bufferMode) {
   const FuncScope funcScope(*this, "transformFeedbackVaryings");
   if (IsContextLost()) return;
@@ -120,12 +120,12 @@
   program.TransformFeedbackVaryings(varyings, bufferMode);
 }
 
-already_AddRefed<WebGLActiveInfo> WebGL2Context::GetTransformFeedbackVarying(
+Maybe<WebGLActiveInfo> WebGL2Context::GetTransformFeedbackVarying(
     const WebGLProgram& program, GLuint index) {
   const FuncScope funcScope(*this, "getTransformFeedbackVarying");
-  if (IsContextLost()) return nullptr;
+  if (IsContextLost()) return Nothing();
 
-  if (!ValidateObject("program", program)) return nullptr;
+  if (!ValidateObject("program", program)) return Nothing();
 
   return program.GetTransformFeedbackVarying(index);
 }